packages/core/src/base/cacheManagerKeyv.ts

```diff
@@ -0,0 +1,339 @@
+    }
+
+    // Get the cached response body
+    const body: string = await this.keyv.get(getContentKey(req));
+
+    return Promise.resolve(
+      new fetch.Response(Buffer.from(body), {
+        url: req.url,
+        headers: resHeaders,
+        status: 200,
+        size: redisInfo.size
+      })
+    );
+  }
+
+  /**
+   * Takes both a request and its response and adds it to the given cache.
+   *
+   * @param req The request this is a storing a response for
+   * @param response The response to be stored
+   * @param opts
+   *
+   * @returns The response or copy of the response
+   */
+  async put(
+    req: fetch.Request,
+    response: fetch.Response,
+    opts?: any
+  ): Promise<fetch.Response> {
+    opts = opts || {};
+    if (!req) {
+      throw new Error("Valid request object required to put");
+    }
+    if (!response) {
+      throw new Error("Valid response object required to put");
+    }
+    this.stripUncacheableRequestHeaders(req);
+    const size = response?.headers?.get("content-length");
+    const metadataKey = getMetadataKey(req);
+    const contentKey = getContentKey(req);
+    const ttl = getTimeToLive(response);
+
+    if (ttl === 0) {
+      return response;
+    }
+
+    const cacheOpts = {
+      algorithms: opts.algorithms,
+      integrity: null,
+      metadata: {
+        url: req.url,
+        reqHeaders: req.headers.raw(),
+        resHeaders: response.headers.raw()
+      },
+      size,
+      time: Date.now()
+    };
+    if (req.method === "HEAD" || response.status === 304) {
+      // Update metadata without writing
+      const redisInfo = await this.keyv.get(metadataKey);
+      // Providing these will bypass content write
+      cacheOpts.integrity = redisInfo.integrity;
+      addCacheHeaders(
+        response.headers,
+        "",
+        contentKey,
+        redisInfo.integrity,
+        redisInfo.time
+      );
+      await this.keyv.set(metadataKey, cacheOpts, ttl);
+
+      await this.updateTimeToLiveForKey(contentKey, ttl);
+
+      return response;
+    }
+
+    const body = await response.text();
+
+    await this.keyv.set(metadataKey, cacheOpts, ttl);
+
+    await this.keyv.set(contentKey, body, ttl);
+
+    return Promise.resolve(new fetch.Response(Buffer.from(body), response));
+  }
+
+  /**
+   * Finds the Cache entry whose key is the request, and if found, deletes the
+   * Cache entry and returns a Promise that resolves to true. If no Cache entry
+   * is found, it returns false.
+   *
+   * @param req The request to try to delete a cached response for
+   * @param opts Compatibility with make-fetch-happen, currently unused
+   *
+   * @returns true is anything is present to delete, false otherwise
+   */
+  async delete(req: fetch.Request, opts?: any): Promise<boolean> {
+    if (!req) {
+      throw new Error("Valid request object required to delete");
+    }
+    this.stripUncacheableRequestHeaders(req);
+    return (
+      (await this.keyv.delete(getMetadataKey(req))) ||
```

Comment:
is || not a short circuit evaluation? If both metadata and content are present in cache, don't the if return true as soon as metadata is deleted

Reply:
good catch! fixed